Fix NativeCallable test issue and make it work x-plat

yizhang82 opened the pull request:
I've tested on windows and on mac and verify they both return 100.
@tijoytom @botaberg PTAL

tijoytom commented:
@yizhang82 It fails on windows since it can't find NativeCallableDll, check the cmake projectname

yizhang82 commented:
@tijoytom The problem seems that in CMakeList.txt we are not copying the DLL to the right folder. It should be: 
install (TARGETS NativeCallableDll DESTINATION Interop/NativeCallable) as opposed to  
install (TARGETS NativeCallableDll DESTINATION bin) 
I just fixed this and squashed the submit. 
It seems that all our interop tests have the same issue, but I can't explain why it worked for them. It seems that Loader/NativeLib also follows this pattern.

yizhang82 commented:
@tijoytom I finally figured this out. We need to add reference to CMakeList.txt from the .csproj and then the right thing would happen automatically to copy the native DLL from bin folder to test directory, which makes sense. 
@tijoytom @botaberg We should make sure we are following this pattern.

janvorli commented on tests/src/Interop/NativeCallable/NativeCallableTest.csproj:
```diff
@@ -37,5 +37,9 @@
   </ItemGroup>
+  <ItemGroup>
+    <!-- This is needed to make sure native binary gets installed in the right location -->
+    <ProjectReference Include="CMakeLists.txt" />
```
I am not sure I understand how adding a cmake project file reference to the msbuild project helps. I can imagine that the dependency would trigger the build of the managed part if the cmakelists.txt changed, but I don't see how it would ensure the native part installation.

yizhang82 replied:
@ramarag @sergiy-k @adityamandaleeka I heard from Sergiy that there is some magic being done to the scripts to copy the binaries and Rama/Aditya might know. Otherwise, I can dig a deeper into this.

ramarag replied:
By giving the relative path to the cmake file in the project is as good as taking a project dependency on *.nativeproj in msbuild. There is back ground logic which translates the relative path to bring in the native files built by the project into the csproj's current directory

yizhang82 replied:
@ramarag That would make a lot of sense. Thanks

tijoytom commented:
@dotnet-bot test  OSX x64 Checked Build and Test

dnfclas commented:
__@yizhang82__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, DNFBOT;
